### PR TITLE
Fix EOF Bug

### PR DESCRIPTION
**Pull request by m36dot**
remove ability to consume EOF token at parse top level node

Fix : #23547 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -503,7 +503,7 @@ private STNode parseCompUnit() {
         boolean processImports = true;
         while (token.kind != SyntaxKind.EOF_TOKEN) {
             STNode decl = parseTopLevelNode(token.kind);
-            if (decl.kind == SyntaxKind.IMPORT_DECLARATION) {
+            if (decl != null && decl.kind == SyntaxKind.IMPORT_DECLARATION) {
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:506]: Shall we do this null check at the beginning and break the loop?

Comment (m36dot) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:506]: sure

**Comment by SupunS** [comment]
Shall we add a test case?

File: compiler/ballerina-parser/src/test/resources/declarations/func-definition/func_def_source_19.bal [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import ballerina/io;
+public function main() {
+   io:println("Hello World!");
+}
+
+func
```
Comment (SupunS) [inline compiler/ballerina-parser/src/test/resources/declarations/func-definition/func_def_source_19.bal:6]: Missing new line

**Pull request merged**